azure-communication-ui/chat/src/main/java/com/azure/android/communication/ui/chat/service/sdk/ChatPollingHandler.kt

```diff
@@ -0,0 +1,301 @@
+                                    eventType = ChatEventType.CHAT_MESSAGE_EDITED.into(),
+                                    infoModel = model,
+                                    eventReceivedOffsetDateTime = null
+                                )
+                                eventSubscriber(infoModel)
+                            }
+
+                            // new message
+                            if (message.deletedOn == null && message.editedOn == null) {
+                                val model = MessageInfoModel(
+                                    internalId = null,
+                                    id = message.id,
+                                    messageType = message.type.into(),
+                                    version = message.version,
+                                    content = message.content.message,
+                                    senderCommunicationIdentifier = message.content.initiatorCommunicationIdentifier?.into(),
+                                    senderDisplayName = message.senderDisplayName,
+                                    createdOn = message.createdOn,
+                                    deletedOn = null,
+                                    editedOn = null
+                                )
+                                val infoModel = ChatEventModel(
+                                    eventType = ChatEventType.CHAT_MESSAGE_RECEIVED.into(),
+                                    infoModel = model,
+                                    eventReceivedOffsetDateTime = null
+                                )
+                                eventSubscriber(infoModel)
+                            }
+                        }
+
+                        if (message.type == ChatMessageType.TOPIC_UPDATED) {
+                            val model = ChatThreadInfoModel(
+                                receivedOn = message.editedOn ?: message.createdOn,
+                                topic = message.content.topic
+                            )
+                            val infoModel = ChatEventModel(
+                                eventType = ChatEventType.CHAT_THREAD_PROPERTIES_UPDATED.into(),
+                                infoModel = model,
+                                eventReceivedOffsetDateTime = null
+                            )
+                            eventSubscriber(infoModel)
+                        }
+                    }
+                }
+            }
+        } catch (ex: Exception) {
+            // TODO: notify sdk wrapper about error in future
+        }
+    }
+
+    /**
+     * RingBuffer uses a fixed length array to implement a queue, where,
+     * - [tail] Items are added to the tail
+     * - [head] Items are removed from the head
+     * - [capacity] Keeps track of how many items are currently in the queue
+     */
+    internal class RingBuffer<T>(private val maxSize: Int = 0) {
```

Comment:
Can Deque be used instead?
https://docs.oracle.com/javase/7/docs/api/java/util/Deque.html

Reply:
Adam, it seems like initialCapacity?

Reply:
I think it accepts initial limit that increases itself?

Reply:
removed as per today discussion.